### ISSUE TITLE
The Dimensionless dimension selection results in values with " one" string appended to them

### ISSUE THREAD

**Issue by spikeygg** [eclipse/smarthome#5445]
I have found that when using the "Dimensionless" option on items that result in a Number, the resulting value is the numeric value followed by the string " one". This post was suggested as the result from the following forum post on the openhab forum: https://community.openhab.org/t/new-homeseer-wd200-switches-coming/39244/46. I have found that I can use the "Area" dimension as a replacement which will not apply any units and still allow for numerical comparisons.

Here is an example from my openhab2 events.log:

```
2018-04-21 23:43:32.083 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from NULL to 0.87 one
2018-04-21 23:43:43.487 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from 0.87 one to 0.86 one
2018-04-21 23:44:06.525 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from 0.86 one to 0.87 one
2018-04-21 23:44:29.452 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from 0.87 one to 0.88 one
2018-04-21 23:44:52.600 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from 0.88 one to 0.87 one
2018-04-21 23:45:04.150 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from 0.87 one to 0.86 one
2018-04-21 23:45:15.614 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from 0.86 one to 0.85 one
2018-04-21 23:45:27.163 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from 0.85 one to 0.84 one
```

When I updated the dimension from "Dimensionless" to "Area" the events.log file appeared like this:
```
2018-04-22 21:54:07.698 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from 0.73 one to 0.74 one
2018-04-22 21:54:19.195 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from 0.74 one to 0.73 one
2018-04-22 21:54:30.659 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from 0.73 one to 0.72 one
2018-04-22 21:54:41.952 [ome.event.ItemUpdatedEvent] - Item 'ZM_Server_CPULoad' has been updated.
2018-04-22 21:54:42.156 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from NULL to 0.71
2018-04-22 21:54:53.865 [vent.ItemStateChangedEvent] - ZM_Server_CPULoad changed from 0.71 to 0.7

```

A dimensionless item should not have any units.

**Comment by maggu2810** [comment]
Sounds like a bug to me.

**Comment by htreu** [comment]
The default "unit" for the dimension "Dimensionless" is actually the abstract unit "ONE". This unit renders with the "one" prefix. I did notice that in Paper UI its not possible to unset the dimension for a Number item. Bit my guess is that this is the bug, not the rendering.
A NumberItem which is bound to a channel only providing DecimalType should not define a dimension.

**Comment by htreu** [comment]
Hi @spikeygg what do you mean by

> and still allow for numerical comparisons

Are you using the item ZM_Server_CPULoad in a rule? Using "Dimensionless" for this item seems to be correct and I just realised that we may have a bug which prevents you from numerical comparison in a rule or script against simple numbers.

Please give some more context what is not working for you. Thanks.

**Comment by spikeygg** [comment]
> Are you using the item ZM_Server_CPULoad in a rule? Using "Dimensionless" for this item seems to be correct and I just realised that we may have a bug which prevents you from numerical comparison in a rule or script against simple numbers.

Hello htreu, not with ZM_Server_CPULoad but with a switch in my house. I just purchased a [HomeSeer HS-WD200+](https://www.amazon.com/gp/product/B079F38TPF). It has some optional triggers such as two-clicks up, four-clicks down, etc. The switch reports the following for each case ([from mcqwerty's post](https://community.openhab.org/t/new-homeseer-wd200-switches-coming/39244/34?u=spikeygg)):

>`[Paddle Direction] . [Action]`
>
>[Paddle Direction]
>1.n = Top
>2.n = Bottom
>
>[Action]
>n.0 = 1 Click
>n.1 = Release
>n.2 = Hold
>n.3 = 2 Clicks
>n.4 = 3 Clicks
>n.5 = 4 Clicks
>n.6 = 5 Clicks
>
>e.g.
>1.3 = Top, 2 Clicks
>2.0 = Bottom, 1 Click

- When configuring these I thought "Dimensionless" would fit well because I would want to compare the value directly: 1.5 == 1.5.
- When I tried to do so, the case statement would not work.
- When I realized (in the event.log file) that it was appending the " one" to the output, I wondered if that was why.
- When I changed the dimension of the value to "Area" my case function and script worked correctly.

At this point, I made the assumption that it wasn't working because of the appended " one" due to the Dimensionless selection. To be honest, I was surprised to hear you say that the "Dimensionless" option is **supposed** to render the value with a _prefix_ of "one" -- that doesn't make sense to me and I would not understand why anyone would expect that to be the case when choosing something called "Dimensionless".

For reference, the beginning the [Dimensionless quantity wiki article](https://simple.wikipedia.org/wiki/Dimensionless_quantity), states:
>In dimensional analysis, a dimensionless quantity is a quantity to which no physical dimension is applicable. It is also known as a bare number or a quantity of dimension one and the corresponding unit of measurement in the SI is one (or 1) unit and it is not explicitly shown.

It doesn't really matter to me since I was able to work around the issue by using the "Area" dimension but there really should be some way to verify what exactly these Dimension selections will *do* to your input value before you go about selecting them. Looking through the huge list of dimensions doesn't give me any clue as to what will end up appending/adding to the output.

Let me know if you have any other questions.

Thank you,
-Greg

**Comment by htreu** [comment]
Thanks greg, thats a sufficient explanation. The discussion in #5453 has some additional technical details, but I think we actually will drop the "one" prefix and in addition allow for numerical comparison of scalar values against the ONE unit or all Dimensionless units.
Also the de-selection of the dimension in Paper UI should be possible, in your case a dimension is not required on the Number item (but you can not de-select it).
Be aware though that when we fixed this issues you should re configure your setup to Dimensionless or unset the dimension. Once a default unit for Area gets configured your setup will be broken again. Stay tuned and thanks for the bug report!

**Comment by htreu** [comment]
Reopen, because https://github.com/eclipse/smarthome/pull/5453 does not fix the issue when comparing scalar and unit ONE values.

**Comment by kaikreuzer** [comment]
Now fully fixed by #5468.
